dotCMS/src/main/java/com/dotmarketing/portlets/rules/RuleComponentDefinition.java

```diff
@@ -26,7 +31,20 @@
     private final Map<String, ParameterDefinition> parameterDefinitions;
 
     protected RuleComponentDefinition(String i18nKey, ParameterDefinition... parameterDefinitions) {
-        this.id = this.getClass().getSimpleName();
+        this(null, i18nKey, parameterDefinitions);
+    }
+
+    /**
+     * @deprecated Do not use this method directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    protected RuleComponentDefinition(String id, String i18nKey,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'i18nKey' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)